aspnetcore/signalr/authn-and-authz.md

```diff
@@ -103,6 +95,25 @@ var connection = new HubConnectionBuilder()
 
 Windows Authentication is only supported by the browser client when using Microsoft Internet Explorer or Microsoft Edge.
 
+### Using Claims to Customize Identity Handling
+
+If you're using organizational or social authentication in your apps and you'd prefer to have more fine-grained control over how SignalR identifies users, you can implement `IUserIdProvider` to return values from one of the claims collected by the identity provider. 
+
+The code below demonstrates how you would use Claims to select the user's email address as the identifying property. 
+
+[!code-csharp[Email claim-based provider](authn-and-authz/sample/EmailBasedUserIdProvider.cs?range=6-12)]
+
+In the sample code, the account registration code contains the following line of code, which adds a claim with type `ClaimsTypes.Email` to the ASP.NET identity database. 
+
+[!code-csharp[Adding the email to the ASP.NET identity claims](authn-and-authz/sample/pages/account/Register.cshtml.cs?range=64)]
+
+The sample code contains both the `Identity.Name`-specific and email-specific implementations. Register this component in your `Startup.ConfigureServices` method **after** the call to `.AddSignalR`. 
```

Comment:
`Identity.Name`-specific and email-specific implementations. --> name and email implementations.

Reply:
done!